### ISSUE TITLE
processing .enex file with two notes with same title loses one of them

### ISSUE THREAD

**Issue by jht001** [akosbalasko/yarle#512]
I've pasted in the conversion log below.  You can see from the log that there were 6 notes in the enex file and they were all processed, but it used the same output name for two different notes (first and last notes), resulting in 5 output files instead of 6.  I'm using Yarle 5.2 on Windows 10

```
Current config is: {
    "enexSources": [
        "C:\\Users\\James Thompson\\Downloads\\JHT.enex"
    ],
    "outputDir": "C:\\Users\\James Thompson\\Downloads\\test_jht_enex",
    "keepOriginalHtml": false,
    "isMetadataNeeded": true,
    "isNotebookNameNeeded": false,
    "isZettelkastenNeeded": false,
    "useZettelIdAsFilename": false,
    "plainTextNotesOnly": false,
    "skipWebClips": false,
    "useHashTags": true,
    "nestedTags": {
        "separatorInEN": "_",
        "replaceSeparatorWith": "---",
        "replaceSpaceWith": "-"
    },
    "outputFormat": "ObsidianMD",
    "taskOutputFormat": "ObsidianMD",
    "obsidianTaskTag": "",
    "urlEncodeFileNamesAndLinks": false,
    "sanitizeResourceNameSpaces": false,
    "replacementChar": "_",
    "pathSeparator": "/",
    "resourcesDir": "_resources",
    "turndownOptions": {
        "headingStyle": "atx"
    },
    "skipLocation": true,
    "skipCreationTime": false,
    "skipUpdateTime": false,
    "skipSourceUrl": false,
    "skipTags": false,
    "skipEnexFileNameFromOutputPath": false,
    "keepMDCharactersOfENNotes": false,
    "monospaceIsCodeBlock": false,
    "currentTemplate": "---\r\n\r\n{created-at-block}_Created at {created-at}._{end-created-at-block}\r\n{updated-at-block}_Last updated at {updated-at}._{end-updated-at-block}\r\n{source-url-block}_Source URL []({source-url})._{end-source-url-block}\r\n\r\n\r\n{tags-block}\r\nTagged \r\n```\r\n{tags}\r\n```\r\n{end-tags-block}\r\n\r\n---\r\n\r\n{title-block}# {title}{end-title-block}\r\n\r\n\r\n{content-block}{content}{end-content-block}\r\n\r\n",
    "logseqSettings": {
        "journalNotes": false
    },
    "obsidianSettings": {
        "omitLinkDisplayName": false
    },
    "dateFormat": "YYYY-MM-DD",
    "keepImageSize": "ObsidianMD",
    "keepOriginalAmountOfNewlines": false,
    "haveEnexLevelResources": false,
    "haveGlobalResources": false,
    "useUniqueUnknownFileNames": false
}
Path separator:\
path C:\Users\James Thompson\Downloads\test_jht_enex\notes\JHT created
Getting stream from C:\Users\James Thompson\Downloads\JHT.enex


Conversion started at Fri Sep 01 2023 12:26:42 GMT-1000 (Hawaii-Aleutian Standard Time)
Converting note "Untitled Note"...
Note saved to C:\Users\James Thompson\Downloads\test_jht_enex\notes\JHT\Untitled Note.md
Conversion finished at Fri Sep 01 2023 12:26:43 GMT-1000 (Hawaii-Aleutian Standard Time)
Note "Untitled Note" converted successfully in 0.55 seconds.
Notes processed: 1




Conversion started at Fri Sep 01 2023 12:26:43 GMT-1000 (Hawaii-Aleutian Standard Time)
Converting note "Untitled Note"...
Note saved to C:\Users\James Thompson\Downloads\test_jht_enex\notes\JHT\Untitled Note.1.md
Conversion finished at Fri Sep 01 2023 12:26:43 GMT-1000 (Hawaii-Aleutian Standard Time)
Note "Untitled Note" converted successfully in 0.048 seconds.
Notes processed: 2




Conversion started at Fri Sep 01 2023 12:26:43 GMT-1000 (Hawaii-Aleutian Standard Time)
Converting note "Untitled Note"...
Note saved to C:\Users\James Thompson\Downloads\test_jht_enex\notes\JHT\Untitled Note.2.md
Conversion finished at Fri Sep 01 2023 12:26:43 GMT-1000 (Hawaii-Aleutian Standard Time)
Note "Untitled Note" converted successfully in 0.115 seconds.
Notes processed: 3




Conversion started at Fri Sep 01 2023 12:26:43 GMT-1000 (Hawaii-Aleutian Standard Time)
Converting note "Your Receipt"...
relative resource work dir: ./_resources/Your_Receipt.resources
absolute resource work dir: C:\Users\James Thompson\Downloads\test_jht_enex\notes\JHT\_resources\Your_Receipt.resources
mediaReference src ./_resources/Your_Receipt.resources/logo.gif added
mediaReference src ./_resources/Your_Receipt.resources/pp.gif added
mediaReference src ./_resources/Your_Receipt.resources/visa.gif added
mediaReference src ./_resources/Your_Receipt.resources/31.gif added
mediaReference src ./_resources/Your_Receipt.resources/receipt_sl.gif added
mediaReference src ./_resources/Your_Receipt.resources/amex_protect.gif added
mediaReference src ./_resources/Your_Receipt.resources/amex.gif added
mediaReference src ./_resources/Your_Receipt.resources/mc.gif added
mediaReference src ./_resources/Your_Receipt.resources/review_mo.gif added
mediaReference src ./_resources/Your_Receipt.resources/dsc.gif added
mediaReference src ./_resources/Your_Receipt.resources/ssl.gif added
mediaReference src ./_resources/Your_Receipt.resources/payship_mo.gif added
Note saved to C:\Users\James Thompson\Downloads\test_jht_enex\notes\JHT\Your Receipt.md
Conversion finished at Fri Sep 01 2023 12:26:43 GMT-1000 (Hawaii-Aleutian Standard Time)
Note "Your Receipt" converted successfully in 0.182 seconds.
Notes processed: 4




Conversion started at Fri Sep 01 2023 12:26:43 GMT-1000 (Hawaii-Aleutian Standard Time)
Converting note "CDEpack"...
relative resource work dir: ./_resources/CDEpack.resources
absolute resource work dir: C:\Users\James Thompson\Downloads\test_jht_enex\notes\JHT\_resources\CDEpack.resources
mediaReference src ./_resources/CDEpack.resources/cde-python-plot-500.png added
mediaReference src ./_resources/CDEpack.resources/bob-1.png added
mediaReference src ./_resources/CDEpack.resources/pg-at-cs.gif added
mediaReference src ./_resources/CDEpack.resources/cde-openscenegraph-class-project.png added
mediaReference src ./_resources/CDEpack.resources/bob-2.png added
mediaReference src ./_resources/CDEpack.resources/cruzer_usb.jpg added
mediaReference src ./_resources/CDEpack.resources/github-blue.png added
mediaReference src ./_resources/CDEpack.resources/packrat.jpg added
mediaReference src ./_resources/CDEpack.resources/alice-2.png added
mediaReference src ./_resources/CDEpack.resources/datacenter_lg.jpg added
mediaReference src ./_resources/CDEpack.resources/cde-google-earth-500.png added
mediaReference src ./_resources/CDEpack.resources/Bug.png added
mediaReference src ./_resources/CDEpack.resources/alice-1.png added
mediaReference src ./_resources/CDEpack.resources/cde-ompl-screenshot.png added
Note saved to C:\Users\James Thompson\Downloads\test_jht_enex\notes\JHT\CDEpack.md
Conversion finished at Fri Sep 01 2023 12:26:44 GMT-1000 (Hawaii-Aleutian Standard Time)
Note "CDEpack" converted successfully in 0.458 seconds.
Notes processed: 5




Conversion started at Fri Sep 01 2023 12:26:44 GMT-1000 (Hawaii-Aleutian Standard Time)
Converting note "Untitled note"...
relative resource work dir: ./_resources/Untitled_note.resources
absolute resource work dir: C:\Users\James Thompson\Downloads\test_jht_enex\notes\JHT\_resources\Untitled_note.resources
mediaReference src ./_resources/Untitled_note.resources/unknown_filename.png added
Note saved to C:\Users\James Thompson\Downloads\test_jht_enex\notes\JHT\Untitled note.md
Conversion finished at Fri Sep 01 2023 12:26:44 GMT-1000 (Hawaii-Aleutian Standard Time)
Note "Untitled note" converted successfully in 0.048 seconds.
Notes processed: 6


==========================
Conversion finished: 6 succeeded, 0 skipped, 0 failed. Total notes: 6
```

**Comment by akosbalasko** [comment]
Hi @jht001 ,

Hmm, it's super-strange, I don't really understand how it can happen. Could you please send me the problematic enex file in order to reproduce the problem? 
Here is my email address: akos0215@gmail.com
Thanks a lot, and sorry for the inconviences! 
akos

**Comment by jht001** [comment]
Hi,
Thanks for following up.
Yarle is a great program to get info out of Evernote!
There is some private info in some of the notes in my enex file, so I don't
want to send the full file.
But I think I may have found the problem.
I've not used typescript before, but it looks like the getFileIndex
function in src/utils/filename-utils.ts  supposed to detect duplicate file
names:

If you look at the yarle conversion log I posted, and look specifically at
the note titles and the markdown file names
The first note:

Converting note "Untitled Note"...
Note saved to C:\Users\James
Thompson\Downloads\test_jht_enex\notes\JHT\Untitled Note.md

The last note:

Converting note "Untitled note"... [this note has exactly this title in
Evernote]
relative resource work dir: ./_resources/Untitled_note.resources
absolute resource work dir: C:\Users\James
Thompson\Downloads\test_jht_enex\notes\JHT\_resources\Untitled_note.resources
mediaReference src
./_resources/Untitled_note.resources/unknown_filename.png added
Note saved to C:\Users\James
Thompson\Downloads\test_jht_enex\notes\JHT\Untitled note.md

The two file names: "Untitled Note.md" and "Untitled note.md"  are the same
file on the Windows case insensitive filename system.
As a result, the last note overwrites the first note.
I think this means that other files generated based on the note name
(resources) can also overwrite each other when this happens.
Any two notes names with the same title except for their case (like "test
abc" and "test ABC") would have the same issue.

Let me know if you agree.
Thanks.
Jim

**Comment by akosbalasko** [comment]
Hi @jht001 ,

oo, okay, it totally makes sense, I check how to resolve it safely asap. 

Cheers,
akos

**Comment by akosbalasko** [comment]
HI @jht001 ,
Thanks for raising this issue, I released a fixed version (5.5.2), please take a look at it. 
Regards,
akos

**Comment by jht001** [comment]
I tried 5.5.2 and it fixed the issue, Thank you!
There is another problem caused by the windows file system rules for valid file names.  I'll post it as a separate issue.
Thanks again.
